src/main/java/de/uni_hamburg/informatik/swt/se2/kino/ui/barzahlung/BarzahlungView.java

```diff
@@ -0,0 +1,207 @@
+ */
+public class BarzahlungView
+{
+    // Die Widgets, aus denen das UI sich zusammensetzt
+    private JDialog _dialog;
+    private JTextField _einzahlungTextField;
+    private JLabel _restbetragLabel;
+    private JLabel _preisLabel;
+    private JButton _bezahlenButton;
+    private JButton _abbrechenButton;
+    private JLabel _preisWertLabel;
+
+    /**
+     * Initialisert die Oberfläche. Die Parameter sind die Views der Submodule,
+     * die eingebettet werden.
+     */
+    public BarzahlungView(Geldbetrag geldbetrag)
+    {
+        _dialog = new JDialog();
+        _dialog.setDefaultCloseOperation(JDialog.DISPOSE_ON_CLOSE);
+        _dialog.setResizable(false);
+        _dialog.setModal(false);
+        _dialog.getContentPane()
+            .setLayout(new BorderLayout());
+
+        JComponent topPanel = erstelleUeberschriftPanel();
+        JComponent centerPanel = erstelleCenterPanel(geldbetrag);
+        JComponent bottomPanel = erstelleBeendenPanel();
+
+        _dialog.getContentPane()
+            .add(topPanel, BorderLayout.NORTH);
+        _dialog.getContentPane()
+            .add(centerPanel, BorderLayout.CENTER);
+        _dialog.getContentPane()
+            .add(bottomPanel, BorderLayout.SOUTH);
+    }
+
+    /**
+     * Zeigt das Fenster an.
+     */
+    public void zeigeFenster()
+    {
+        _dialog.setSize(400, 300);
+        _dialog.setVisible(true);
+    }
+
+    /**
+     * Schließt das Fenster.
+     */
+    public void schliesseFenster()
+    {
+        _dialog.dispose();
+    }
+
+    private JPanel erstelleUeberschriftPanel()
+    {
+        JPanel topPanel = new JPanel();
+        JLabel label = new JLabel("Barzahlung", SwingConstants.CENTER);
+
+        Font font = label.getFont()
+            .deriveFont(Font.BOLD + Font.ITALIC, 20);
+        label.setFont(font);
+        label.setForeground(Color.blue);
+
+        topPanel.setLayout(new BorderLayout());
+        topPanel.add(label, BorderLayout.CENTER);
+
+        return topPanel;
+    }
+
+    /**
+     * Erzeugt das Panel mit der Überschrift fuer das Programm.
+     */
+    private JPanel erstelleCenterPanel(Geldbetrag geldbetrag)
+    {
+        JPanel centerPanel = new JPanel();
+        centerPanel.setLayout(new BorderLayout());
+        
+        _preisLabel = new JLabel("Fehlermeldung", SwingConstants.CENTER);
+        JComponent barzahlungPanel = erstelleBarzahlungPanel(geldbetrag);
+
+        centerPanel.add(_preisLabel, BorderLayout.SOUTH);
+        centerPanel.add(barzahlungPanel, BorderLayout.CENTER);
+
+        return centerPanel;
+    }
+
+    /**
+     * Erzeugt das Panel mit der Überschrift fuer das Programm.
+     */
+    private JPanel erstelleBarzahlungPanel(Geldbetrag geldbetrag)
+    {
+        JPanel barzahlungPanel = new JPanel();
+        barzahlungPanel.setLayout(new GridLayout(2, 3));
+        JLabel textPreisLabel = new JLabel("Preis", SwingConstants.CENTER);
+        _preisWertLabel = new JLabel(geldbetrag.toString(),
```

Comment:
Durch das mehrfache Durchreichen von Geldbetrag wird hier ein Währungszeichen, in unserem Falle € vergessen! Lieber direkt im Controller machen, denn um die Währung soll sich die View nicht kümmern